bin/yb-ctl

```diff
@@ -1331,20 +1429,50 @@ class ClusterControl:
         info_kv_list = []
         if not tserver_daemon_id:
             return info_kv_list
+
+        tserver_ip_address = tserver_daemon_id.get_ip_address()
+
+        # -----------------------------------------------------------------------------------------
+        # ysqlsh command line
+        # -----------------------------------------------------------------------------------------
+
         if self.is_ysql_enabled():
-            ysql_port = self.options.base_ports[tserver_daemon_id.daemon_type]["ysql_rpc"]
+            ysql_port = self.options.get_ysql_port()
+            ysqlsh_cmd_line = self.options.get_client_tool_path('ysqlsh')
+            if tserver_ip_address != LOCALHOST_IP:
+                ysqlsh_cmd_line += ' -h %s' % tserver_ip_address
+            if ysql_port != YSQL_DEFAULT_PORT:
+                ysqlsh_cmd_line += ' -p %s' % ysql_port
             info_kv_list.extend([
                 ("JDBC", "postgresql://postgres@{}:{}".format(
-                    tserver_daemon_id.get_ip_address(), ysql_port)),
-                ("YSQL Shell", "./bin/ysqlsh")
+                    tserver_ip_address, ysql_port)),
+                ("YSQL Shell", ysqlsh_cmd_line)
             ])
+
+        # -----------------------------------------------------------------------------------------
+        # cqlsh command line
+        # -----------------------------------------------------------------------------------------
+
+        ycql_port = self.options.get_ycql_port()
+        cqlsh_cmd_line = self.options.get_client_tool_path('cqlsh')
+        if tserver_ip_address != LOCALHOST_IP or ycql_port != YCQL_DEFAULT_PORT:
+            cqlsh_cmd_line += ' %s' % tserver_ip_address
+            if ycql_port != YCQL_DEFAULT_PORT:
+                cqlsh_cmd_line += ' %s' % ycql_port
+
         info_kv_list.append(
-            ("YCQL Shell", "./bin/cqlsh"))
+            ("YCQL Shell", cqlsh_cmd_line))
+
+        # -----------------------------------------------------------------------------------------
+        # redis-cli command line
+        # -----------------------------------------------------------------------------------------
+
         # TODO: should probably not even display this if we can know it's not there...
-        yedis_port = self.options.base_ports[tserver_daemon_id.daemon_type]["yedis_rpc"]
-        info_kv_list.append(
-            ("YEDIS Shell", "./bin/redis-cli -h {} -p {}".format(
-                tserver_daemon_id.get_ip_address(), yedis_port)))
+        yedis_port = self.options.get_yedis_port()
+        redis_cli_cmd_line = "{} -h {} -p {}".format(
```

Comment:
does `redis-cli` have to have host and port always, or why are we not doing the same special treatment here too?